Loop/Managers/LoopDataManager.swift

```diff
@@ -1507,6 +1507,24 @@ extension LoopState {
     func predictGlucose(using inputs: PredictionInputEffect, includingPendingInsulin: Bool = false) throws -> [GlucoseValue] {
         try predictGlucose(using: inputs, potentialBolus: nil, potentialCarbEntry: nil, replacingCarbEntry: nil, includingPendingInsulin: includingPendingInsulin)
     }
+    
+    /// Indicates if the provided the insulin counteraction effects threshold was reached by an average insulin counteraction effect calculated over the provided time frame.
+    /// - Parameter threshold: The insulin counteraction effects threshold
+    /// - Parameter startDate: The earliest date to filter the insulin counteraction effects
+    /// - Parameter endDate: The latest date to filter the insulin counteraction effects
+    /// - Returns: a boolean indicating if the threshold was surpassed
+    func didInsulinCounteractionEffectsReachThreshold(_ threshold: HKQuantity, startDate: Date, endDate: Date) -> Bool {
+        let filteredInsulinCounteractionEffects = insulinCounteractionEffects.filterDateRange(startDate, endDate)
```

Comment:
Accessing `insulinCounteractionEffects` should only be done on the `dataAccessQueue`